### PR TITLE
Remove `cvd::Request` from `cvd` internal APIs

### PR DESCRIPTION
**Pull request by Databean**
[`cvd::Request`](https://github.com/google/android-cuttlefish/blob/16694cc91c52ca7d9121fd8f8a56b2af0ddb5f71/base/cvd/cuttlefish/host/commands/cvd/cvd_server.proto#L33) is no longer relevant now that `cvd` doesn't have a client/server split, and there is no RPC boundary.

Bug: b/369474354

### CODE REVIEW

File: base/cvd/cuttlefish/host/commands/cvd/server_client.h [hunk 1]
```diff
@@ -19,54 +19,101 @@
...
   std::ostream& Err() const;
 
   bool IsNullIo() const;
 
-  // Convenient accessors to commonly used properties in the underlying message
-  cvd_common::Args Args() const {
-    return cvd_common::ConvertToArgs(Message().command_request().args());
+  // Convenient getters/setters to properties in the underlying message
+  RequestWithStdio& OverrideRequest(RequestWithStdio) &;
+  RequestWithStdio OverrideRequest(RequestWithStdio) &&;
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_client.h:48]: What's the difference between these two and `operator=`?

File: base/cvd/cuttlefish/host/commands/cvd/server_client.cpp [hunk 2]
```diff
@@ -122,4 +87,92 @@ bool RequestWithStdio::IsNullIo() const {
...
+  return *this;
+}
+
+RequestWithStdio RequestWithStdio::AddSelectorArgument(
+    std::string argument) && {
+  selector_args_.emplace_back(std::move(argument));
+  return *this;
+}
+
+RequestWithStdio& RequestWithStdio::ImportEnv(const cvd_common::Envs& env) & {
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_client.cpp:134]: nit: `ExtendEnv`?

File: base/cvd/cuttlefish/host/commands/cvd/server_client.cpp [hunk 3]
```diff
@@ -122,4 +87,92 @@ bool RequestWithStdio::IsNullIo() const {
...
+  for (const auto& [key, value] : env) {
+    env_[key] = value;
+  }
+  return *this;
+}
+
+RequestWithStdio RequestWithStdio::ImportEnv(const cvd_common::Envs& env) && {
+  for (const auto& [key, value] : env) {
+    env_[key] = value;
+  }
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_client.cpp:144]: Could this method just call the `&` version instead of repeating the `for` loop?

Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/server_client.cpp]: I've removed `ImportEnv`. It was used in an intermediate commit, but now `SetEnv` works as a substitute.

File: base/cvd/cuttlefish/host/commands/cvd/server_client.cpp (hunk 2 above)
Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/server_client.cpp]: I've removed `ImportEnv`. It was used in an intermediate commit, but now `SetEnv` works as a substitute.

File: base/cvd/cuttlefish/host/commands/cvd/server_client.h (hunk 1 above)
Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/server_client.h]: The difference is it doesn't reassign the `istream` and `ostream`s. It was only being called in an intermediate commit, so I've removed it.

File: base/cvd/cuttlefish/host/commands/cvd/cvd_server.proto
[the remark this answers is not in the document]
Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/cvd_server.proto:1]: Done.

**Review by jemoreira — APPROVED** [review]

File: base/cvd/cuttlefish/host/commands/cvd/acloud/converter.cpp [hunk 4]
```diff
@@ -375,29 +373,29 @@ Result<ConvertedAcloudCreateCommand> ConvertAcloudCreate(
...
-    fetch_command.add_args("--directory");
-    fetch_command.add_args(host_dir);
-    fetch_command.add_args("--default_build");
+    RequestWithStdio& fetch_request =
+        inner_requests.emplace_back(RequestWithStdio::InheritIo(request))
+            .AddArgument("cvd")
+            .AddArgument("fetch")
+            .AddArgument("--directory")
+            .AddArgument(host_dir)
+            .AddArgument("--default_build");
```
[the remark this answers is not in the document]
Comment (Databean) [reply base/cvd/cuttlefish/host/commands/cvd/acloud/converter.cpp]: Replaced `.AddArgument(x).AddArgument(y)` with `.AddArguments({x, y})`.

File: base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp [hunk 5]
```diff
@@ -181,64 +180,44 @@ class LoadConfigsCommand : public CvdServerHandler {
...
+            /* cvd load will always create instances in daemon mode (to be
+             independent of terminal) and will enable reporting automatically
+             (to run automatically without question during launch)
+             */
+            .AddArguments({"cvd", "create", "--daemon", system_build_arg})
+            .AddArguments(cvd_flags.launch_cvd_flags)
+            .AddSelectorArguments(cvd_flags.selector_flags)
+            .AddSelectorArguments({"--group_name", group.GroupName()});
+
+    auto& env = launch_req.Env();
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp:211]: This environment is empty, on the previous version it was the one on the request.

File: base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp [hunk 6]
```diff
@@ -145,8 +144,7 @@ Result<StatusFetcherOutput> StatusFetcher::FetchOneInstanceStatus(
                                             .args = cmd_args,
                                             .envs = envs,
                                             .working_dir = working_dir,
-                                            .command_name = bin,
-                                            .null_stdio = request.IsNullIo()};
+                                            .command_name = bin};
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp:147]: the `.null_stdio` field is missing

File: base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp [hunk 7]
```diff
@@ -181,64 +180,44 @@ class LoadConfigsCommand : public CvdServerHandler {
...
+    if (Contains(env, kAndroidProductOut)) {
+      env.erase(kAndroidProductOut);
     }
 
-    // Make sure the newly created group is used by cvd create
-    launch_cmd.mutable_selector_opts()->add_args("--group_name");
-    launch_cmd.mutable_selector_opts()->add_args(group.GroupName());
-
-    return RequestWithStdio::InheritIo(std::move(launch_req), request);
+    return request;
```
Comment (jemoreira) [inline base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp:220]: `return launch_req`

File: base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp (hunk 5 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp:212]: Done

File: base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp (hunk 6 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp]: Done

File: base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp (hunk 7 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/server_command/load_configs.cpp]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cvd.cpp [hunk 8]
```diff
@@ -64,16 +64,16 @@ Result<cvd::Response> Cvd::HandleCommand(
...
     const std::unordered_map<std::string, std::string>& env,
     const std::vector<std::string>& selector_args) {
-  cvd::Request request = MakeRequest({.cmd_args = cvd_process_args,
-                                      .env = env,
-                                      .selector_args = selector_args});
+  RequestWithStdio request = RequestWithStdio::StdIo()
+                                 .AddArguments(cvd_process_args)
+                                 .SetEnv(env)
+                                 .AddSelectorArguments(selector_args)
+                                 .SetWorkingDirectory(CurrentDirectory());
```
[the remark this answers is not in the document]
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cvd.cpp:71]: no need to fix

**Review by jemoreira — APPROVED** [review]

**Pull request merged**